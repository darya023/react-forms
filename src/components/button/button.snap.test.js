import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router';
import {createMemoryHistory} from 'history';
import Button from './button';

let history;
describe(`Tests for Button`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });
  it(`Button should render correctly`, () => {
    const {container} = render(
        <Router history={history}>
          <Button>
            Test
          </Button>
        </Router>
    );
    expect(container).toMatchSnapshot();
  });
  it(`Button should render correctly with type submit`, () => {
    const {container} = render(
        <Router history={history}>
          <Button type="submit">
            Test
          </Button>
        </Router>
    );
    expect(container).toMatchSnapshot();
  });
  it(`Button should render correctly if disabled`, () => {
    const {container} = render(
        <Router history={history}>
          <Button disabled={true}>
            Test
          </Button>
        </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
