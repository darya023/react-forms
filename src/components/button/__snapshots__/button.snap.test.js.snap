// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button Button should render correctly 1`] = `
<div>
  <button
    class="button"
  >
    Test
  </button>
</div>
`;

exports[`Tests for Button Button should render correctly if disabled 1`] = `
<div>
  <button
    class="button"
    disabled=""
  >
    Test
  </button>
</div>
`;

exports[`Tests for Button Button should render correctly with type submit 1`] = `
<div>
  <button
    class="button"
    type="submit"
  >
    Test
  </button>
</div>
`;
